### PR TITLE
DAT-3992 | Infobox Builder: ask for template title on save attempt

### PR DESCRIPTION
**Pull request by dianafa**
## Links

https://wikia-inc.atlassian.net/browse/DAT-3992

## Reviewers

@hakubo @rogatty

### CODE REVIEW

File: front/main/app/router.js [hunk 1]
```diff
@@ -22,6 +22,10 @@ Router.map(function () {
 		path: '/infobox-builder/*templateName'
 	});
 
+	this.route('infobox-builder', {
+		path: '/infobox-builder/'
```
Comment (RafLeszczynski) [inline front/main/app/router.js]: we don't need path here

File: front/main/app/routes/infobox-builder.js [hunk 2]
```diff
@@ -98,12 +98,18 @@ export default Ember.Route.extend(ConfirmationMixin, {
 		 * @returns {Ember.RSVP.Promise}
 		 */
 		redirectToTemplatePage() {
+			let action = 'redirectToPreviousPage';
```
Comment (RafLeszczynski) [inline front/main/app/routes/infobox-builder.js]: what do you think about:
```JS
const action = this.controllerFor('infobox-builder').get('model.title') ? 'redirectToTemplatePage' : 
````

**Comment by SebastianMarzjan** [comment]
Please resolve conflicts

File: front/main/app/components/infobox-builder.js [hunk 3]
```diff
@@ -277,6 +328,35 @@ export default Ember.Component.extend(
...
+		 */
+		showEditTitleModal(trigger) {
+			track({
+				action: trackActions.open,
+				category: 'infobox-builder',
+				label: `edit-title-modal-before-triggered-on-${trigger}`
+			});
+
+			this.set('editTitleModalTrigger', trigger);
+			this.set('isEditTitleModalVisible', true);
```
Comment (rogatty) [inline front/main/app/components/infobox-builder.js]: Please use `setProperties` when setting more than one.

File: front/main/app/templates/components/infobox-builder.hbs [hunk 4]
```diff
@@ -73,9 +73,24 @@
 			ns='infobox-builder'}}</p>
 		{{/pop-over}}
 	</div>
-	{{#modal-dialog isVisible=showGoToSourceModal class='infobox-builder-go-to-source-modal'}}
-		{{infobox-builder-go-to-source-modal goToSource=(action 'goToSource')
-		}}
+	{{#if canGoToSourceModal}}
```
Comment (RafLeszczynski) [inline front/main/app/templates/components/infobox-builder.hbs]: do we need this? what about using `set()` for computed property and passing this value to `isVisible`?

File: front/common/public/locales/en/infobox-builder.json [hunk 5]
```diff
@@ -31,6 +31,13 @@
...
     "source-editor-save-confirmation-button-no": "No",
-    "source-editor-loading": "Loading source editor"
+    "source-editor-loading": "Loading source editor",
+    "untitled-infobox-template": "Untitled Infobox Template",
+    "edit-title-placeholder": "Untitled template",
+    "edit-title-modal-header": "Name your template",
+    "edit-title-modal-text": "To publish your template you have to provide name for it.",
+    "edit-title-modal-publish": "Publish",
+    "edit-title-modal-ok": "Ok",
+    "edit-title-modal-cancel": "Cancel"
```
Comment (SebastianMarzjan) [inline front/common/public/locales/en/infobox-builder.json]: Could we sort the keys? Or doesn't it make sense?

File: front/main/app/components/infobox-builder.js (hunk 3 above)
Comment (RafLeszczynski) [inline front/main/app/components/infobox-builder.js]: good catch :+1:

File: front/main/app/components/infobox-builder.js [hunk 6]
```diff
@@ -45,6 +53,24 @@ export default Ember.Component.extend(
 			return Boolean(this.get('activeItem')) && !this.get('isPreviewItemDragged');
 		}),
 
+		infoboxTemplateTitle: Ember.computed('title', function () {
+			return this.get('title') || i18n.t('infobox-builder:main.untitled-infobox-template');
+		}),
+
+		editTitleModalConfirmButtonLabel: Ember.computed('editTitleModalTrigger', function () {
+			const messageKey = this.get('editTitleModalTrigger') === 'publish' ?
+				'save' :
```
Comment (SebastianMarzjan) [inline front/main/app/components/infobox-builder.js]: A side note - the message keys could use more love

File: front/main/app/components/infobox-builder.js [hunk 7]
```diff
@@ -45,6 +58,24 @@ export default Ember.Component.extend(
...
+				'edit-title-modal-save' :
+				'edit-title-modal-ok';
+
+			return i18n.t(`main.${messageKey}`, {
+				ns: 'infobox-builder'
+			});
+		}),
+
+		showEditTitleModalCancelButton: Ember.computed('editTitleModalTrigger', function () {
+			return this.get('editTitleModalTrigger') !== 'publish';
```
Comment (SebastianMarzjan) [inline front/main/app/components/infobox-builder.js]: shouldn't it be 'edit-title-modal-publish' ?

File: front/main/app/routes/infobox-builder.js [hunk 8]
```diff
@@ -155,8 +159,7 @@ export default Ember.Route.extend(ConfirmationMixin, {
 					if (data && data.isWikiaContext && data.isLoggedIn) {
 						resolve();
 					} else {
-						// @todo DAT-3757 show message that user doesn't have proper permission
-						// currently this else block isn't called even for anons
+						reject('Builder launched not in Wikia context');
```
Comment (SebastianMarzjan) [inline front/main/app/routes/infobox-builder.js]: Suggestion: 'not in' => 'outside of'

File: front/main/app/components/infobox-builder.js (hunk 7 above)
Comment (dianafa) [inline front/main/app/components/infobox-builder.js]: yep, `editTitleModalTrigger` is an action name, so stays `publish`

**Comment by SebastianMarzjan** [comment]
To me looks good, however conflict I sense

File: front/main/app/styles/component/_infobox-builder.scss [hunk 9]
```diff
@@ -342,11 +352,48 @@ $sidebar-z-index: 1;
...
+	}
+
+	.modal-text-area {
+		padding: $modal-text-padding $modal-text-padding $modal-text-bottom-padding;
 	}
+
+	.modal-bottom-row {
+		border-top: $border-width $color-blue-gray-light solid;
+		text-align: right;
+		padding: $modal-bottom-row-padding;
```
Comment (bognix) [inline front/main/app/styles/component/_infobox-builder.scss]: alphabetical order

File: front/main/app/templates/components/infobox-builder-edit-title-modal.hbs [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+		name='editTemplateTitle'
+		onFocusHandler=(action 'onInputFocus')
+	}}
+</div>
+<div class="modal-bottom-row">
+	<button {{action this.attrs.confirmButtonAction value}} disabled={{isConfirmButtonDisabled}}>
+		{{confirmButtonText}}
+	</button>
+	{{#if showCancelButton}}
+		<button {{action this.attrs.cancelButtonAction}}>{{cancelButtonText}}</button>
```
Comment (bognix) [inline front/main/app/templates/components/infobox-builder-edit-title-modal.hbs]: `this` is redundant

**Comment by bognix** [comment]
LGTM - minor notes

**Pull request merged**